packages/react-components/react-combobox/src/utils/Selection.types.ts

```diff
@@ -14,11 +14,11 @@ export type SelectionProps = {
   multiselect?: boolean;
 
   /* Callback when an option is selected */
-  onSelect?(event: SelectionEvents, data: OnSelectData): void;
+  onOptionSelect?: (event: SelectionEvents, data: OptionOnSelectData) => void;
```

Comment:
This has the `On` in the middle to align with other form component data types like `InputOnChangeData`, probably going to keep this as-is (also based on my other comment on the event name. Otherwise I suppose `DropdownOnOptionSelectData` would be possible :P)